tools/walletextension/frontend/src/components/providers/wallet-provider.tsx

```diff
@@ -189,16 +189,26 @@ export const WalletConnectionProvider = ({
       setProvider(providerInstance);
       initialize(providerInstance);
 
-      ethereum.on("accountsChanged", fetchUserAccounts);
+      const handleAccountsChanged = async (accounts: string[]) => {
+        if (accounts.length === 0) {
+          setAccounts(null);
+          setWalletConnected(false);
+          setToken("");
+        } else {
+          window.location.reload();
```

Comment:
ive had this cause issues with wallet connections in the past so you might wanna consider using `setAccounts(accounts)` and `setWalletConnected(true)` and then using that as a hook?